pkg/mount/nfs.go

```diff
@@ -66,11 +70,35 @@ func (m *nfsMounter) Reload(source string) error {
 		return fmt.Errorf("Internal error failed to convert %T",
 			newNFSmounter)
 	}
+	var newM *Info
+	if m.handleDNSResolution {
+		// Check if the source is a IP:share combination which maps to an
+		// DNS:share combination.
+		inputSourceExportPath := extractSourcePath(inputSource)
+		for existingSource, existingMountInfo := range newNFSmounter.mounts {
+			if inputSourceExportPath == extractSourcePath(existingSource) {
+				// Found a match for the same export path.
+				// Now lets check if the input source (IP or DNS) matches
+				// with the existing source (IP or DNS).
+				resolvedExistingSourceIPs := resolveToIPs(existingSource)
+				resolvedInputSourceIPs := resolveToIPs(inputSource)
```

Comment:
Nit: resolvedInputSourceIPs can be done outside the loop. Anyways since path is matching, we should be ok.